Source/WebKit/UIProcess/API/glib/WebKitImagePrivate.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#include "WebKitImage.h"
+#include <WebKit/WKBase.h>
+#include <wtf/CompletionHandler.h>
+#include <wtf/glib/GRefPtr.h>
+
+WK_EXPORT WebKitImage* webkitImageNew(int width, int height, guint stride, CompletionHandler<GRefPtr<GBytes>(WebKitImage*)>&& createBytesFn);
```

Comment:
It's difficult for me to understand why we need the completion handler, without seeing how this is used from webkit.

Reply:
That patch doesn't look simple to me, rather the opposite, the completion handler makes it complex for no reason. Since this is internal API, I think it's better to keep it simple and if we ever need to create an image in a different way we can change it. For now it can just receive a GBytes instead of a completion handler.

Reply:
I'm removing the completion handler for now, passing bytes directly